### ISSUE TITLE
Support for M5Stack Core2 planned?

### ISSUE THREAD

**Issue by kraus-wagner** [sieren/Homepoint#145]
Will you support the new M5Stack Core2?

**Comment by sieren** [comment]
Hey, the Core2 should already be supported. 
All it needs is to add `"displayColorInverted": true` to the configuration file since it's display is wired up a bit differently.

**Comment by cerietke** [comment]
Interesting, not sure you're aware but the Dutch magazine Computer Totaal mentioned your work and they suggested the Core2 implementation wasn't yet ready (https://computertotaal.nl/artikelen/internet-thuis/zo-maak-je-een-eenvoudig-dashboard-voor-je-domoticasysteem/). This is why I got myself a Core. The Core2 with its touchscreen would be way cooler.

I actually found that to get the colors in their article I also had to set the displayColorInverted to true on the Core.

**Comment by sieren** [comment]
Oh, wow. I wasn't aware of this! I guess it's time I order myself a Core2 and tidy this up nicer!

**Comment by cerietke** [comment]
Ah, cool. I actually found their description of how to get things up and running pretty helpful at times, it felt slightly more detailed than your own (their demographic includes less experienced users as well). I'm a Product Owner, not a die-hard techy, though I like fiddling around with things, possibly it would be useful if I write some thing up regarding my experience as feedback?

**Comment by sieren** [comment]
Absolutely! There's a Wiki here on Github which I sadly have neglected way too much where this could be very useful for!

**Comment by cerietke** [comment]
Where do I send it?

**Comment by koenvervloesem** [comment]
> Ah, cool. I actually found their description of how to get things up and running pretty helpful at times, it felt slightly more detailed than your own

Thanks, that was the intention of my article, to help non-techies with getting Homepoint up and running :-) Nice to hear that this worked.

In fact, I tried flashing Homepoint to the M5Stack Core2, but the device didn't boot (or at least I didn't see anything), and I didn't have the time then to investigate this further. Maybe the display code needs more changes than just inverting the colors, or maybe it's something else. In the mean time, I have been using my M5Stack Core 2 for another purpose, so I haven't been able to look at it further.

But I agree with the touch screen it would be a very cool device for Homepoint!

**Comment by sieren** [comment]
@koenvervloesem Thanks for the article! :)
I will probably end up creating a special _bin file for M5Stack Core2 after all to make it easier for newcomers.
If I had unlimited resources I'd probably even get rid of the JSON Editor in favour of a modular Web UI.

@cerietke you can mail it to me at matthias (at) s-r-n.de

**Comment by sieren** [comment]
Mijn nederlands is unfortunately not zo goed als dat ik dit en engels vertaalen kan (als en duitser ;)).
I think everyone would probably benefit from these step by step instructions. Super happy to host them and put them prominently on the page first.
I'll make sure to link to the article in the Wiki as well.

**Comment by kraus-wagner** [comment]
@koenvervloesem Good article! I read it with google translator. ;-)
@sieren Great project!

I was looking for a touchscreen for an ESP32 or ESP8266 to send mqtt signals to my mosquitto and ioBroker.
It should fit in a wall switch. A common system size in Germany is 55mm (Gira...).
So i found the M5Stack Core2. A perfect gadget. ESP32, touchscreen, perfect size.
First steps in UIflow and Arduino IDE followed. Then i found your project. It is so amazing.
Everything can be controlled by a fingertip with mqtt. I am sure i will love it when it works on the Core2.

**Comment by cerietke** [comment]
Cool that you're also checking in here Koen, I was not aware of homepoint nor the M5Stacks and it covers a lot of nice usecases I was looking for solutions for.

Better stick with English or German ;-). I'll mail you with feedback.

As an experiment I essentially turned the M5Stack Core into a lights and music remote control for my bedroom (receiver is in another room), but I was also thinking about the wall switch replacement option.

**Comment by sieren** [comment]
Thanks! 

For Wallswitches, I vaguely recall there was a Kickstarter Project trying to achieve this too:
https://www.kickstarter.com/projects/159751026/23-smarthome-touch-display-w-wlan-for-55mm-switch-series

**Comment by kraus-wagner** [comment]
Yes i found that project too. But the Core2 is ready to go. I ordered 2 from digikey.de for 39 Euro each.
https://www.digikey.de/product-detail/de/m5stack-technology-co-ltd/K010/2221-K010-ND/13151126
If your order is over 50 Euro shipping is free. They ship from USA with UPS in 3-4 days. Customs is in the price included.
So i paid 78 Euro total for 2 Core2. Shipping to my door.

**Comment by sieren** [comment]
Received one today, but this will take longer than expected. Can't even get the screen to work right now.
And I need to write a new touch driver from scratch for this

**Comment by kraus-wagner** [comment]
I use the Arduino IDE. There i have to change the library from "#include <M5Stack.h>" to "#include <M5Core2.h>" for example to run most code for the Stack on the Core2. I dont know how it works with the Espressif IDE.

**Comment by sieren** [comment]
Yeah. But HomePoint supports multiple platforms with multiple capabilities and introducing dependencies on specific platforms is to be avoided. For it to work on all of them the code has to be generic and abstracted enough, and there come the challenges when a device introduces new components (like Core2 using a different touch-screen handler and moving certain TFT related capabilities to a separate power management chip)

**Comment by kraus-wagner** [comment]
Ok i see. It is much more work than changing a library. But i hope you like your Core2 and will have time to make homepoint run on it.
✅

**Comment by cerietke** [comment]
I got one as well, so hoping to see homepoint work on it at some point in the future.
Paid 34+2.50 shipping at tinytronics.nl.

**Comment by sieren** [comment]
![IMG_1440](https://user-images.githubusercontent.com/5174440/107068230-a300b580-67e0-11eb-9e43-35fbb0bc33d5.jpg)

Got the display up and running today, next up Touch Driver support. 

If you feel like helping out but lack the programming skills. the best way is a small donation (there's a link at the end of the front page README) as I bought this device just to add compatibility for all the requests.
Personally I don't really have that much need for it.

**Comment by jpiscaer** [comment]
Yay! Sitting here at virtual FOSDEM and tinkering with HomePoint and the Core2. Display won't turn on, version 0.6 with displayColorInverted set. Going to wait for a pre-compiled binary core the Core2.

**Comment by sieren** [comment]
[homepoint_release.zip](https://github.com/sieren/Homepoint/files/5937726/homepoint_release.zip)

Here's a first beta. Sadly the images dont look super pretty. The Core2 library is sadly also a far-cry from the Core1 one (https://github.com/m5stack/M5Core2/issues/33). Ultimately I might have to tear the Core2 one apart, which might mean lots of more work until things look as good as with other ESP32 devices.

**Comment by kraus-wagner** [comment]
I uploaded the new core2 binary and have access to the web interface. I change the config file, so the Core2 is in my wifi network.
But the screen stays dark. I added "displayColorInverted": "true", to the config file and rebooted. Screen stays dark.
Any tips?

**Comment by sieren** [comment]
Sorry. there was a bug in the build script that leads to the binary actually being the same as M5Stack (non-Core2)
[homepoint_release.zip](https://github.com/sieren/Homepoint/files/5939033/homepoint_release.zip)

**Comment by kraus-wagner** [comment]
Now it works! I made an OTA update and a reboot. 
"displayColorInverted": "true", must be removed from config if previously has been added. 
Now i can do a little testing ;-)

**Comment by sieren** [comment]
Alright, just implemented proper JPG drawing, so no more artefacts. Did a new release too: https://github.com/sieren/Homepoint/releases/tag/v0.07

If you enjoy this work, please consider donating so I might be able to recoup the Core2 which I bought for this ;)

**Comment by jpiscaer** [comment]
Trying to flash 0.07, but after flashing the full (and the OTA), I hear the device constantly booting with a typical 'reset' tick sound. It physically makes noise: ticking and sometimes a high-pitch whining. Screen won't turn on, and no 'homepoint' access point. If I flash back the factory test program with m5burner, the device works. Any idea what's happening? Factory test indicates no hardware problems.
`esptool.py write_flash -fs 4MB -fm dout 0x0 homepoint_m5stackcore2_ota_update.bin`

**Comment by kraus-wagner** [comment]
@jpiscaer 

Hi, i first flashed the core2_full binary.
Then connected via the browser to 192.168.99.1 and entered my wifi settings. 
Next reconnection via the new ip in my network.
Last i made the OTA update also via the browser frontend.
(screen stays dark during all the time)
After a reboot the screen should work.

**Comment by sieren** [comment]
Oh boy, this looks like a bug. I'll take a look later today.

**Comment by sieren** [comment]
Phew, sorry for the annoyance: https://github.com/sieren/Homepoint/releases/tag/v0.07.1

**Comment by jpiscaer** [comment]
Yes, this one boots. Thank you Matt! I appreciate your hard work

**Comment by cerietke** [comment]
I had difficulty with connecting to it over WiFi. Both initially for the WiFi config as well as after being able to connect and set it up. I found sometimes it would help to restart. It seems flashing it a 3rd time it's stable as I have been able to load my config. Is there some way to confirm if there might be something up with my hardware?

P.s. donation made

**Comment by sieren** [comment]
Mhmm, not sure. Maybe with the Factory tool it came with? 
I haven't had any issues with it so far. It does sound a bit dubious though, I haven't made any changes to the WIFI Part of it at all, its the same across all esp32 builds.

Thank you for the donation! appreciate it a lot 🙏☺️

**Comment by jpiscaer** [comment]
i have seen the same behavior consistently: in the first couple of minutes after boot, WiFi is unstable, but it stabilizes afterwards. WiFi confirmed to be working with other firmware and factory test.

**Comment by sieren** [comment]
I assume you mean it's crashing when you talk about "unstable" ? Thats a bit vague for me, but i did see some reboots when setting it up initially just now. I'll investigate.

**Comment by jpiscaer** [comment]
Yes, with unstable I meant reboots, plus the dropping of connection in the first minutes after boot (seen by the wifi icon on the display)

**Comment by dummylabs** [comment]
I can also confirm that m5core2 freezes after a couple of hours with the Homepoint running on it.

**Comment by heartkingz** [comment]
Great project! I'm trying to use Homepoint but i am also experiencing WiFi issues. Most of the time it won't connect, it is very random. Sometimes a reload helps, sometimes a reboot and after that it connects very fast and i can use it without any issues. What i have noticed is that when you go to the web interface of the device it connects fairly fast after that.. 
Another thing i'm facing is that you have to press pretty long before a button responds, is that by design? I'd love to see a possibility to simply press short. Thank you again it's a great project!

**Comment by sieren** [comment]
Just a quick update on this. I've picked up work on https://github.com/sieren/Homepoint/issues/155 ... it'll take a while but all platforms will benefit from this. I've started with the Core2 as the base development platform before expanding to the other ones. So far everything is stable, but rewriting the whole UI will take a bit more time.

Will keep you posted.


https://user-images.githubusercontent.com/5174440/115853174-2e293900-a429-11eb-84f7-91644a59c19c.MOV

**Comment by sieren** [comment]
https://user-images.githubusercontent.com/5174440/117072109-4570fc00-ad30-11eb-8a80-f7f6234a55b0.mov

Expect a first beta in the coming days, the only thing currently missing is the sensor views and a few refinements before i can put a binary in this thread for some testing. Pretty happy with the new UI and being able to swipe between pages.

**Comment by sieren** [comment]
Here's a very first beta of HomePoint 2.0.
Most of the original functionality should exist. Navigation between screen through swiping, long press to enter a scene with multiple devices, single tap to activate the whole scene.
Top right to exit a scene. Settings button currently does nothing.
Overall still rough around the edges.

Flash Example: `esptool.py --baud 921600 --port /dev/cu.usbserial-0202E890 write_flash -fs 4MB -fm dout 0x0 homepoint_m5stackcore2_full.bin`


[homepoint_m5stackcore2_full.bin.zip](https://github.com/sieren/Homepoint/files/6441172/homepoint_m5stackcore2_full.bin.zip)

**Comment by heartkingz** [comment]
Just flashed the latest bin file tot the core2. Looks really good! I have to play a bit more. 

may I be so free to make suggestions:

- I'd love to see a battery indicator (percentage)/charging state on top 
- IP address to the settings icon top left corner 
- Maybe separator lines between the different rooms 

Again, thank you for your work!

**Comment by sieren** [comment]
Thanks for your feedback. I'll definitely keep that in mind. It's still a long way of cleaning up and refining the current beta before I can think of adding new features. Let me know if you run into some weird or off behaviors.

**Comment by cerietke** [comment]
Seems to work much better than the last version I tried a few months ago. The suggested command didn't work for me, but I got it loaded with "esptool.py write_flash -fs 4MB -fm dout 0x0 D:\Homepoint\homepoint_m5stackcore2_full.bin".

- Did something change with the icon functionality? Can't seem to get it to show my own icons the way it does on my core 1 with older installation. I used to just upload them through the web interface.
- Loving the scrolling between screens, much better, also better for a mix of core 1 and core 2 as the screens will both be able to contain 6 items.
- First connection for config was a bit slow, but after that it seems to be working fine.
- I may need to look into my config, but everything I have on it seems to be a group rather than a button. Definitely assuming that's on me.
- Would be nice if "back" took you back to the screen you were on.

**Comment by sieren** [comment]
Yeah, the images now need to be on a fixed 16 bit color scale and be converted to binary files through https://lvgl.io/tools/imageconverter. The ending is now ".bin", loading and decoding JPGs/PNGs on the fly was just too slow, especially with the animations.

Entering a group is through a single tap, long-pressing triggers the whole group.
You can leave the screen by tapping on the top (little X), but I need to tweak this a little, its definitely too fiddly on the small Core2 screen.

**Comment by cerietke** [comment]
I just noticed, if you have 6 devices at the lower level you only have the tiny top left 'x' to get back up, can be hard to hit. Was just going to post that, but I see you're aware. How about swipe up to go back?

Can I have a top level device that I just toggle on/off without going into the group?

Did the  "screenSaverMinutes": 1, setting stop working?

Are these the correct settings for the images? Or, where do I get that fixed 16 bit color scale from?
![image](https://user-images.githubusercontent.com/78350158/119236024-76756d00-bb35-11eb-89ed-c56a06e72b43.png)

**Comment by sieren** [comment]
Maybe I'll keep the back button at the end. Might be worth it.
Detecting Swipe Gestures (ie. swipe up) isn't trivial with lvgl, but i'll take a look..

Yes, I guess you can long-press it for now to toggle it on/off, but I might add some logic to ensure a single tap is sufficient if the group only contains a single device, good point!

**Comment by cerietke** [comment]
I'd much prefer no back button as it gives an extra space. Obviously I don't know how difficult the swipe up could be. The icon could certainly be a bit bigger. Or how about clicking anywhere on the top bar to go up?

Btw, talking about icon sizes, the wifi one seems to end up a little too far to the right?

The 1 device group logic would work for my scenario, or possibly a way to specify adding a device at the top level that isn't a group? Might not need extra logic then and may prove more flexible?

I'd really miss the screensaver, ideally I'd be able to put it at like 5 or 10 seconds. Also, using those settings I was unable to create an image file the core2 could work with, hence the question how I could get that fixed 16 bit color scale?

**Comment by fvanroie** [comment]
You should be able to detect the [swipe gesture and direction](https://github.com/lvgl/lvgl/issues/1650#issuecomment-668594714) in your event handler like this:

```cpp
void event_handler(lv_obj_t* obj, lv_event_t event)
{
    if(event == LV_EVENT_GESTURE) {
        lv_gesture_dir_t dir = lv_indev_get_gesture_dir(lv_indev_get_act());
        switch(dir) {
            case LV_GESTURE_DIR_LEFT:
                break;
            case LV_GESTURE_DIR_RIGHT:
                break;
            case LV_GESTURE_DIR_BOTTOM:
                break;
            default:
                break; // TOP
        }
    } else {
      // other events
    }
}
```

**Comment by sieren** [comment]
Awesome..I'll add this. But will keep the x at the top too. As nice as this gesture is, discovery might be an issue.

**Comment by sieren** [comment]
@cerietke you have to make sure your images are of PNG & 16 bit indexed colors.
Homepoint 2.0 is using black (000000) as a chroma-keyed background, meaning black will become transparent.

Here are a two images from the upcoming release:
![exitbutton](https://user-images.githubusercontent.com/5174440/119256539-6efba580-bbc1-11eb-90d2-0ce689984499.png)
![livingroom](https://user-images.githubusercontent.com/5174440/119256541-702cd280-bbc1-11eb-882b-813b0deeca26.png)

Worth noting that re-coloring these is now happening on the fly. Eliminating the need to have two separate icons.

**Comment by cerietke** [comment]
Ha, I started from JPG. Still unsure how to get paint.net to output as 16 bit indexed, but I'll try later and report back.

**Comment by cerietke** [comment]
Though the results look quite terrible I have been able to get it to display an image. 

![image](https://user-images.githubusercontent.com/78350158/119261359-50080e00-bbd7-11eb-99c7-8ba7b1d54764.png)

This came out as dark grey and black.

**Comment by sieren** [comment]
Yes, because it's saved as RGB and not Indexed Colors.

Try running this through the converter:
![119261359-50080e00-bbd7-11eb-99c7-8ba7b1d54764](https://user-images.githubusercontent.com/5174440/119264953-5604eb80-bbe5-11eb-9d49-99d8b5ea4fda.png)

**Comment by cerietke** [comment]
I'm sure it can be done. But what do you use to get to the correct image? I cannot find the correct option in paint.net. I am happy to use something else. Or am I using the wrong settings on https://lvgl.io/tools/imageconverter ?

**Comment by sieren** [comment]
Both GIMP and Photoshop can get the job done. (e.g. https://docs.gimp.org/2.10/en/gimp-image-convert-indexed.html )

Your settings seemed right to me

**Comment by cerietke** [comment]
Indeed appears like paint.net doesn't have an indexed mode.

**Comment by cerietke** [comment]
Trying gimp, when I copy your file it works. When I start from my png it does not.

I do image -> mode -> indexed -> maximum number of colors (16) -> convert, then I export to png (leaving the default settings). There must be something you're doing different than I am.

Besides the images, any insight on the screensaver? I'd plan to use this as a nightstand device, so screen always on would be a no-go. In fact it would be very nice to be able to disable the LED at the bottom and any other lights as well.

**Comment by sieren** [comment]
Yes, I converted it to Greyscale before going indexed just to make sure there's no colors left in it.
Could be worth a shot?

As for the screensaver: It's coming back at some point. It was a lot of work transitioning to this library and I'm not done yet, I'll do the screensaver pretty soon after cause I'm facing the same problem as you.

**Comment by sieren** [comment]
Oh, one thing I forgot. I used the "Fill" tool once on the black background of your image since it wasnt really entirely black before. there were a few greyish pixels.

**Comment by cerietke** [comment]
Tried grayscale -> indexed, also tried with a save in between. Still doesn't show the picture.

I also see a black / white option in indexed, I am thinking to go for that to make absolutely sure it's only black and white.

![image](https://user-images.githubusercontent.com/78350158/119326989-cca3f680-bc82-11eb-9e7e-b9317342da49.png)

**Comment by sieren** [comment]
What do you mean with "it doesnt show the picture" ? like, not at all?

**Comment by cerietke** [comment]
Yeah, not at all. I also tried to update the json to use it instead of the working picture you provided to confirm it was not a problem with the json.

**Comment by sieren** [comment]
When you switch over to Indexed, chose Colormap of Image or something, it should be 16 colors in total.

<img width="495" alt="Screenshot 2021-05-24 at 11 27 04" src="https://user-images.githubusercontent.com/5174440/119327410-3de3a980-bc83-11eb-8256-992728ead0fb.png">

**Comment by sieren** [comment]
> Tried grayscale -> indexed, also tried with a save in between. Still doesn't show the picture.
> 
> I also see a black / white option in indexed, I am thinking to go for that to make absolutely sure it's only black and white.
> 
> ![image](https://user-images.githubusercontent.com/78350158/119326989-cca3f680-bc82-11eb-9e7e-b9317342da49.png)

This one is an RGB Image. Not Indexed.

**Comment by cerietke** [comment]
I literally chose indexed though :(

**Comment by sieren** [comment]
How do you save these? I opened yours in GIMP and saved it and afterwards double-checked in Photoshop and then it was Indexed.
File -> Export As with the following options:

<img width="403" alt="Screenshot 2021-05-24 at 11 33 23" src="https://user-images.githubusercontent.com/5174440/119327970-dbd77400-bc83-11eb-8322-66fbc83ec545.png">

**Comment by cerietke** [comment]
I can't choose color map of the image under custom palette.

Btw, if you want to communicate more directly I am happy to. I imagine that a process for others being able to create the images would be important.

I left that at the default settings, looks the same as yours, could it be a Mac vs. Windows issue?

![image](https://user-images.githubusercontent.com/78350158/119328105-0c1f1280-bc84-11eb-9250-246f3097cc97.png)

**Comment by kraus-wagner** [comment]
@sieren 

Hi Matt,

i tested the new version and it is cool. Swiping is nice. The config file bug seems fixed.

But every 5 minutes my device restarts?

And i have a problem to get temperatures shown. Switching lights works so mqtt seems to work.

topic : kw97/laden_temperatur/tele/SENSOR
{"Time":"2021-05-23T16:15:51","DS18B20":{"Id":"011927068657","Temperature":14.6},"TempUnit":"C"}

Homepoint config:
"type": "singleValue",
"firstIcon": "temperature_small",
"jsondata": true,
"firstKey": "Temperature",
"getTopic": "kw97/laden_temperatur/tele/SENSOR"

Doesn't work. That was the same in the old version, no new problem.

**Comment by sieren** [comment]
Hey @kraus-wagner, can you paste the whole config? 

@cerietke I've just checked with GIMP and using the same workflow in it I've been using in Photoshop results in no icon showing. (Plus GIMP is an absolute dumpster fire...). I'll continue to investigate this.

**Comment by cerietke** [comment]
Good to know I am not completely crazy 😉. At least with paint.net I did get an image to show. I'm not a big fan of gimp either, possibly paint.net is more promising?

**Comment by sieren** [comment]
<img width="531" alt="Screenshot 2021-05-24 at 18 40 22" src="https://user-images.githubusercontent.com/5174440/119379259-8faa2500-bcbf-11eb-9cdf-0620845db119.png">

Maybe? I suppose Paint.Net is what's formerly known as MS Paint? 😅 
I dont know whats wrong, but you can try using Paint, just make sure the background is completely black (000000) and maybe change the fill threshhold so it colors "less black" pixels black too that might have occurred due to prior compression.

FWIW I can't even get your image working with Photoshop anymore. Maybe my brain's too fried today. Can you attach your original image to work with?

**Comment by cerietke** [comment]
No, it's not MS Paint related. It's getpaint.net btw.

I'm on mobile now, can upload the image tomorrow.

**Comment by kraus-wagner** [comment]
{
  "wifi": "XXX",
  "password": "XXX",
  "login": "XXX",
  "webpass": "XXX",
  "mqttbroker": "mqtt://XXX.XXX.XXX.XXX",
  "mqttusername": "XXX",
  "mqttpasswd": "XXX",
  "timezone": "CET-1CEST-2,M3.5.0/02:00:00,M10.5.0/03:00:00",
  "scenes": 
  [{
    "name": "Esszimmer",
    "type": "Light",
    "icon": "livingroom",
    "devices": 
    [{
      "name": "Stehlampe",
      "setTopic": "kw97/esszimmer_stehlampe/cmnd/POWER",
      "getTopic": "kw97/esszimmer_stehlampe/stat/POWER",
      "onValue": "ON",
      "offValue": "OFF",
      "icon": "bedroom"
    }]
  },
  {
    "name": "Wohnzimmer",
    "type": "Light",
    "icon": "livingroom",
    "devices": 
    [{
      "name": "Stehlampe",
      "setTopic": "kw97/wohnzimmer_stehlampe/power",
      "getTopic": "kw97/wohnzimmer_stehlampe/power",
      "onValue": "1",
      "offValue": "0"
    }]
  },
  {
    "name": "Temperatur 1",
    "type": "Sensor",
    "devices": 
    [{
      "name": "Temperatur 1",
      "type": "singleValue",
      "firstIcon": "temperature_small",
      "jsondata": true,
      "firstKey": "Temperature",
      "getTopic": "kw97/laden_temperatur/tele/SENSOR"
    }]
  },
  {
    "name": "Temperatur 2",
    "type": "Sensor",
    "devices": 
    [{
      "name": "Temperatur 2",
      "type": "combinedValues",
      "firstIcon": "temperature_small",
      "secondIcon": "temperature_small",
      "jsondata": true,
      "firstKey": "DS18B20-1/Temperature",
      "secondKey": "DS18B20-2/Temperature",
      "getTopic": "kw97/pool_steuerung/tele/SENSOR"
    }]
  },
  {
    "name": "Trockner",
    "type": "Sensor",
    "devices": 
    [{
      "name": "Tasmota Plug",
      "type": "singleValue",
      "jsondata": true,
      "firstKey": "Power",
      "firstIcon":"wattage_small",
      "getTopic": "kw97/waschkeller_waeschetrockner/tele/SENSOR"
    }]
  }]
}


All temperatures and power shows 0.
Switching light works.

**Comment by cerietke** [comment]
The image I was starting from:

![next_inactive](https://user-images.githubusercontent.com/78350158/119445992-f9611800-bd2d-11eb-9b05-3ca950049d74.jpg)

I realize this isn't strictly black and white.

**Comment by Girgitt** [comment]
I am not sure if this project is still supported by @sieren but if anyone is interested I made some fixes to the Homepoint v1 (non lvgl-based) version for M5Core2 in my fork: https://github.com/Girgitt/Homepoint/tree/v1-m5core2-fixes 

Major issues fixed: 
1. i2c-related crashes with touch display handling due to rx buffer overflow
2. disabled (for now until I fix it) screensaver which never deactivates

Minor issues fixed: 
1. color conversion from 24b to 16b RRRRRGGGGGGBBBBB format was incorrect


With empty wifi config M5Core2 is still not rock-solid but stable enough to get to the initial config web page to set wifi credentials - just do not refresh the config page multiple times.

With wifi configured you can experience couple crashes on initial access to the config web page but after that it works stable. 

My longest test so far was 38h with about 10h with lost wifi connection (I took it to work; mine has battery in the base) and after getting back home wifi reconnected just fine and mqtt was working straight away. Good enough for me.

**Comment by cerietke** [comment]
If nobody else I definitely appreciate your effort.

**Comment by Girgitt** [comment]
OK, I think I am getting somewhere my dev_1_0_2 build (https://github.com/Girgitt/Homepoint/releases/tag/dev_1_0_2) works ok on three different M5Core2 with multiple pages and hierarchies (room->device). I have not tested measurements, only switching lights. 
The last thing to get to release would be to disable or make configurable behavior for room with devices - I see no real benefit from splitting the already small tile into areas where upper area navigates to devices and bottom area switches all of them. I think the long press should trigger loads in the room and default behaviour should be to get in the the room by tapping the tile no matter where it is tapped. Added feedback (blinking screen and vibrations) helps a bit with tapping but still most of the cases I would open the room instead of switch everything what is there. Configuration should be per tile so one scenario would be covered: switching scenes when the same loads are arranged into scenes and not rooms.

**Comment by Girgitt** [comment]
There is one important prerequisite for M5Core2 to work with Homepoint - the touchscreen internal firmware must be in a correct version. 

Most of my devices (5 out of 6) had firmware version 0x10 which barely works (detecting short presses only, no responsive at all after longer operation). 

The correct version is 0x11. 

I described how to check/change the fw in issue: https://github.com/Girgitt/Homepoint/issues/9
